src/util.ts

```diff
@@ -53,15 +52,55 @@ export function nintendoBase64Encode(decoded: string | Buffer): string {
 	return encoded.replaceAll('+', '.').replaceAll('/', '-').replaceAll('=', '*');
 }
 
+export function generateOAuthTokens(systemType: SystemType, pnid: HydratedPNIDDocument, options?: OAuthTokenOptions): OAuthTokenGenerationResponse {
+	const accessTokenExpiresInSecs = options?.accessExpiresIn ?? 60 * 60; // * 1 hour
+	const refreshTokenExpiresInSecs = options?.refreshExpiresIn ?? 24 * 60 * 60; // * 24 hours
+
+	const accessTokenOptions: TokenOptions = {
+		system_type: systemType,
+		token_type: 'OAUTH_ACCESS',
+		pid: pnid.pid,
+		access_level: pnid.access_level,
+		expire_time: BigInt(Date.now() + (accessTokenExpiresInSecs * 1000)) // * 1 hour
+	};
+
+	const refreshTokenOptions: TokenOptions = {
+		system_type: systemType,
+		token_type: 'OAUTH_REFRESH',
+		pid: pnid.pid,
+		access_level: pnid.access_level,
+		expire_time: BigInt(Date.now() + (refreshTokenExpiresInSecs * 1000)) // * 30 days

```

Comment:
The comments here are kinda misleading since `options` can change this, and also the refresh token time is 1 day by default not 30 anyway?